src/Sarif/Processors/Log/SarifLogStageFactory.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT        
+// license. See LICENSE file in the project root for full license information. 
+
+using Microsoft.CodeAnalysis.Sarif.Visitors;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.Sarif.Processors
+{
+    public class SarifLogProcessorFactory
+    {
+        public static IGenericAction<SarifLog> GetActionStage(SarifLogAction action, params string[] args)
+        {
+            switch (action)
+            {
+                case SarifLogAction.None:
+                    return new GenericMappingAction<SarifLog>(a => a);
+                case SarifLogAction.MakeUrisAbsolute:
+                    return new GenericMappingAction<SarifLog>(log =>
+                    {
+                        AbsoluteUrisVisitor visitor = new AbsoluteUrisVisitor();
+                        return visitor.VisitSarifLog(log);
+                    });
+                case SarifLogAction.RebaseUri:
+                    return new GenericMappingAction<SarifLog>(log =>
+                    {
+                        RebaseUriVisitor visitor = new RebaseUriVisitor(args[0], new Uri(args[1]));
+                        return visitor.VisitSarifLog(log);
+                    });
+                case SarifLogAction.Merge:
+                    return new GenericFoldAction<SarifLog>(mergeFunction);
+                case SarifLogAction.Sort:
+                    throw new NotImplementedException("Future work, not yet supported.");
```

Comment:
>"Future work, not yet supported." [](start = 54, length = 33)

don't need a message here in my opinion, as the exception type communicates everything that the text string does